openff/toolkit/topology/topology.py

```diff
@@ -214,6 +214,107 @@ def __keytransform__(self, key):
         return key
 
 
+class UnsortedDict(_TransformedDict):
+    ...
+
+
+class TagSortedDict(_TransformedDict):
+    """
+    A dictionary where keys are kept unsorted, but only allows one permutation of a key
+    to exist. Certain situations require that atom indices are not transformed in any
```

Comment:
The keys here are tuples of atom indices, correct? I first read this as if the topic was whether or not the keys in the dictionary were sorted, which confused me.

Reply:
Yes, these keys are atom indices. Happy to accept modification if something isn't clear.

Reply:
It probably makes sense to anybody who was already in the headspace of these dictionaries, but I didn't immediately remember what these keys store.